src/caret_analyze/infra/lttng/records_provider_lttng.py

```diff
@@ -1979,6 +2087,76 @@ def inter_comm_records(
             COLUMN_NAME.MESSAGE_TIMESTAMP,
             COLUMN_NAME.SOURCE_TIMESTAMP
         ]
+        if COLUMN_NAME.RMW_TAKE_TIMESTAMP in merged.columns:
+            columns.append(COLUMN_NAME.RMW_TAKE_TIMESTAMP)
+        drop = list(set(merged.columns) - set(columns))
+        merged.drop_columns(drop)
+        merged.reindex(columns)
+
+        # NOTE: After merge, the dropped data are aligned at the end
+        # regardless of the time of publish.
+        merged.sort(COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP)
+
+        return merged
+
+    def inter_take_comm_records(
+        self,
+        publisher_handles: list[int],
+        rmw_handle: int
+    ) -> RecordsInterface:
+        """
+        Compose filtered inter communication records.
+
+        Parameters
+        ----------
+        publisher_handles : list[int]
+            publisher handles
+        rmw_handle : int
+            rmw_take subscription handle
+
+        Returns
+        -------
+        RecordsInterface
+            columns:
+
+            - publisher_handle
+            - rclcpp_publish_timestamp
+            - rmw_take_timestamp
+            - rcl_publish_timestamp (Optional)
+            - dds_write_timestamp (Optional)
+            - message_timestamp
+            - source_timestamp
+
+        """
+        pub_records = self.publish_records(publisher_handles)
+        rmw_records = self._grouped_rmw_take_records[rmw_handle].clone()
+        rmw_records.drop_columns([
+            'rmw_subscription_handle'])
+
+        merged = merge(
+            left_records=pub_records,
+            right_records=rmw_records,
+            join_left_key=COLUMN_NAME.SOURCE_TIMESTAMP,
+            join_right_key=COLUMN_NAME.SOURCE_TIMESTAMP,
+            columns=Columns.from_str(
+                pub_records.columns + rmw_records.columns
+            ).column_names,
+            how='left'
+        )
+
+        columns = [
+            COLUMN_NAME.PUBLISHER_HANDLE,
+            COLUMN_NAME.RCLCPP_PUBLISH_TIMESTAMP,
+            COLUMN_NAME.RMW_TAKE_TIMESTAMP,
+        ]
```

Comment:
The order of the columns does not follow the sequence of event occurrences.
Since inter_take_comm_records() is only called at the end of the path, this is unlikely to cause major issues.
However, for the sake of consistency in the implementation, please reorder the columns to match the order of event occurrences.